### PR TITLE
bpo-33832: Add "magic method" entry to Glossary

### PR DESCRIPTION
**Pull request by andresdelfino**
Proposed merge message.

https://bugs.python.org/issue33832

### CODE REVIEW

File: Doc/reference/datamodel.rst [hunk 1]
```diff
@@ -1118,12 +1118,13 @@ Special method names
 
 A class can implement certain operations that are invoked by special syntax
 (such as arithmetic operations or subscripting and slicing) by defining methods
-with special names. This is Python's approach to :dfn:`operator overloading`,
+with special names, known as "special" or "magic" methods.
+This is Python's approach to :dfn:`operator overloading`,
 allowing classes to define their own behavior with respect to language
-operators.  For instance, if a class defines a method named :meth:`__getitem__`,
+operators.  For instance, if a class defines the special method :meth:`__getitem__`,
```
Comment (pablogsal) [inline Doc/reference/datamodel.rst]: I think these are not the only occurrences of `method` referring to a magic method. A quick grep reveals that there are more places where the construction `the __something__ method` happens. Should we change these too to `__something__ special method`?

**Comment by pablogsal** [comment]
Sorry for being repetitive over other PRs, but could you change the commit messages to include some context or squash them?

File: Doc/reference/datamodel.rst (hunk 1 above)
Comment (andresdelfino) [reply Doc/reference/datamodel.rst]: That's a pretty good question. After doing a couple of replacements, I tried a new grep search and found a lot more. I'm not really sure adding such verbosity helps.

IMHO, having just "method" everywhere, except in Data Model (where the special methods are described) is sufficient, also linking each reference to a special method to Data Model of couse.

The PR right now includes some "method" to "special method" changes. Please let me know what you think about this, and I'll continue with the replacements or revert the changes I made to the PR :)

Comment (andresdelfino) [reply Doc/reference/datamodel.rst]: A grep search shows that "special method" / "magic method" are not used much outside Data model. IMHO, we should keep this as it is. I believe it makes the documentation more readable.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
I think these changes make the doc worse and that this PR should be closed.

**Comment by andresdelfino** [comment]
@terryjreedy the initial commit was lost because I was asked to squash all commits, but it's intention was to just add "magic method" to the Glossary pointing to "special method", and add "magic method" as a synonym in Data model. I can revert this changes so you can see the original intention.

File: Doc/reference/datamodel.rst (hunk 1 above)
Comment (pablogsal) [reply Doc/reference/datamodel.rst]: As I said in the issue: Sorry if this question was interpreted as a change request. It was only me trying to understand the scope of the change (if the proposal intent was to apply or not the change to the rest of the docs).

Comment (andresdelfino) [reply Doc/reference/datamodel.rst]: No problem, Pablo! We are all working to make Python better :)

**Comment by terryjreedy** [comment]
PRs best stick to one subject.  I would be at least neutral to a simple glossary addition.
I don't want to promote the term 'magic method', but it is in use and I suspect various core developers are partly to blame.  So beginners do run across it and wonder how 'magic methods' are different from 'special methods'.  I would be even happier with 'an informal synonym ...'.

Add a blurb: Add glossary entry for 'magic method'.

Please get in the habit of including a blurb if you think one will be needed.  Once present, they are easy to edit through the web interface, without having to pull the PR.

**Comment by andresdelfino** [comment]
I agree with marking the term informal.

Sorry for not adding the blurb. Didn't know it was needed, and have committed a couple of glossary terms with no blurb being requested, but I'll keep it in mind :)

I have made the requested changes; please review again

**Comment by terryjreedy** [comment]
Never squash a PR.  The merge message (commit to master repository message) is ultimately written by the core dev who merges.  But it is nice if the initial commit message under the title is at least an approximation.  It can be edited in, as I just did.

The boilerplate message about backporting should be deleted before hitting the [submit PR] button.

**Comment by andresdelfino** [comment]
Yes, that was my understanding too, but I was asked to squash the commits. I'll have this in mind next time.

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no longer accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

**Comment by merwok** [comment]
I like this addition!

Would one or two index entries also be useful?

**Comment by andresdelfino** [comment]
> I like this addition!
> 
> Would one or two index entries also be useful?

I think so, good idea! I'll add it.

**Comment by andresdelfino** [comment]
@merwok Done! I added index entries for magic method and special method as well.

**Review by merwok — APPROVED** [review]

**Comment by andresdelfino** [comment]
@merwok perhaps we can merge this?

**Pull request merged**